attest/win_events.go

```diff
@@ -0,0 +1,462 @@
+		return fmt.Errorf("reading u%d: invalid varint", header.Size<<8)
+	}
+
+	if w.BootCount > 0 && w.BootCount != int(i) {
+		return fmt.Errorf("conflicting values for boot counter: %d != %d", i, w.BootCount)
+	}
+	w.BootCount = int(i)
+	return nil
+}
+
+func (w *WinEvents) readBitlockerUnlock(header microsoftEventHeader, r io.Reader) error {
+	if header.Size > 8 {
+		return fmt.Errorf("bitlocker data too large (%d bytes)", header.Size)
+	}
+	data := make([]uint8, header.Size)
+	if err := binary.Read(r, binary.LittleEndian, &data); err != nil {
+		return fmt.Errorf("reading u%d: %w", header.Size<<8, err)
+	}
+	i, n := binary.Uvarint(data)
+	if n <= 0 {
+		return fmt.Errorf("reading u%d: invalid varint", header.Size<<8)
+	}
+
+	s := BitlockerStatus(i)
+	if w.seenBitlocker && w.BitlockerStatus != s {
+		return fmt.Errorf("conflicting values for bitlocker status: %v != %v", s, w.BitlockerStatus)
+	}
+	w.BitlockerStatus = s
+	w.seenBitlocker = true
+	return nil
+}
+
+func (w *WinEvents) parseImageBase(header microsoftEventHeader, r io.Reader) (uint64, error) {
+	if header.Size != 8 {
+		return 0, fmt.Errorf("payload was %d bytes, want 8", header.Size)
+	}
+	var num uint64
+	if err := binary.Read(r, binary.LittleEndian, &num); err != nil {
+		return 0, fmt.Errorf("reading u64: %w", err)
+	}
+	return num, nil
+}
+
+func (w *WinEvents) parseAuthenticodeHash(header microsoftEventHeader, r io.Reader) ([]byte, error) {
+	if header.Size > 32 {
+		return nil, fmt.Errorf("authenticode hash data exceeds the size of any valid hash (%d bytes)", header.Size)
+	}
+	data := make([]byte, header.Size)
+	if err := binary.Read(r, binary.LittleEndian, &data); err != nil {
+		return nil, fmt.Errorf("reading bytes: %w", err)
+	}
+	return data, nil
+}
+
+func (w *WinEvents) readLoadedModuleAggregation(rdr io.Reader, header microsoftEventHeader) error {
+	var (
+		r        = io.LimitReader(rdr, int64(header.Size))
+		codeHash []byte
+		imgBase  uint64
+	)
+
+eventLoop:
+	for {
+		var header microsoftEventHeader
+		if err := binary.Read(r, binary.LittleEndian, &header); err != nil {
+			if errors.Is(err, io.EOF) {
+				break eventLoop
+			}
+			return fmt.Errorf("parsing LMA sub-event: %v", err)
+		}
+
+		var err error
+		switch header.Type {
+		case imageBase:
+			if imgBase != 0 {
+				return errors.New("duplicate image base data in LMA event")
+			}
+			if imgBase, err = w.parseImageBase(header, r); err != nil {
+				return err
+			}
+		case authenticodeHash:
+			if codeHash != nil {
+				return errors.New("duplicate authenticode hash structure in LMA event")
+			}
+			if codeHash, err = w.parseAuthenticodeHash(header, r); err != nil {
+				return err
+			}
+		default:
+			return fmt.Errorf("unknown event in LMA aggregation: %v", header.Type)
+		}
+	}
+
+	l := WinModuleLoad{
+		AuthenticodeHash: codeHash,
+		ImageBase:        []uint64{imgBase},
+	}
+	hashHex := hex.EncodeToString(l.AuthenticodeHash)
+	l.ImageBase = append(l.ImageBase, w.LoadedModules[hashHex].ImageBase...)
```

Comment:
need to ensure hashHex is actually in LoadedModules?
```
m, ok := w.LoadedModules[hashHex]
if !ok {
    // ...
}
l.ImageBase = append(l.ImageBase, m.ImageBase...)
```

Reply:
Nack, the `nil` default value of `ImageBase` passed to `append()` should be safe, as `nil...` variadically expands to 0 arguments.